class.jetpack-options.php

```diff
@@ -144,21 +167,9 @@ public static function delete_option( $names ) {
 			delete_option( "jetpack_$name" );
 		}
 
-		$options = get_option( 'jetpack_options' );
-		if ( !is_array( $options ) ) {
-			$options = array();
-		}
-
-		$to_delete = array_intersect( $names, self::get_option_names(), array_keys( $options ) );
-		if ( $to_delete ) {
-			foreach ( $to_delete as $name ) {
-				unset( $options[$name] );
-			}
-
-			return update_option( 'jetpack_options', $options );
-		}
-
-		return true;
+		return
```

Comment:
As slick as this is one-lining it, I worry that it may be a bit overly confusing to some folks reading it.  May be wiser to split it up.